staging/src/k8s.io/apiserver/pkg/server/serve.go

```diff
@@ -58,6 +58,9 @@ func (s *SecureServingInfo) Serve(handler http.Handler, shutdownTimeout time.Dur
 			// enable HTTP2 for go's 1.7 HTTP Server
 			NextProtos: []string{"h2", "http/1.1"},
 		},
+		// TODO: make these configurable
+		ReadTimeout: 10 * time.Second,
```

Comment:
Does this close watch connections after 10 seconds if there’s nothing to read?

Reply:
No

Reply:
Why don't they influence watches? This needs a comment.

Reply:
https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/ 

ReadTimeout covers the time from when the connection is accepted to when the request body is fully read (if you do read the body, otherwise to the end of the headers). It's implemented in net/http by calling SetReadDeadline immediately after Accept.

https://github.com/golang/go/blob/3ba31558d1bca8ae6d2f03209b4cae55381175b3/src/net/http/server.go#L1509